src/spatial_ref/srs.rs

```diff
@@ -303,6 +314,111 @@ impl SpatialRef {
         }
     }
 
+    #[cfg(major_ge_3)]
+    pub fn get_name(&self) -> Result<String> {
+        let c_ptr = unsafe { gdal_sys::OSRGetName(self.0) };
+        if c_ptr.is_null() {
+            return Err(_last_null_pointer_err("OSRGetName"));
+        }
+        Ok(_string(c_ptr))
+    }
+
+    pub fn get_angular_units_name(&self) -> Result<String> {
+        let mut c_ptr = ptr::null_mut();
+        unsafe { gdal_sys::OSRGetAngularUnits(self.0, &mut c_ptr) };
+        if c_ptr.is_null() {
+            return Err(_last_null_pointer_err("OSRGetAngularUnits"));
+        }
+        Ok(_string(c_ptr))
+    }
+
+    pub fn get_angular_units(&self) -> f64 {
+        unsafe { gdal_sys::OSRGetAngularUnits(self.0, ptr::null_mut()) }
+    }
+
+    pub fn get_linear_units_name(&self) -> Result<String> {
+        let mut c_ptr = ptr::null_mut();
+        unsafe { gdal_sys::OSRGetLinearUnits(self.0, &mut c_ptr) };
+        if c_ptr.is_null() {
+            return Err(_last_null_pointer_err("OSRGetLinearUnits"));
+        }
+        Ok(_string(c_ptr))
+    }
+
+    pub fn get_linear_units(&self) -> f64 {
+        unsafe { gdal_sys::OSRGetLinearUnits(self.0, ptr::null_mut()) }
+    }
+
+    #[inline]
+    pub fn is_geographic(&self) -> bool {
+        unsafe { gdal_sys::OSRIsGeographic(self.0) == 1 }
+    }
+
+    #[inline]
+    #[cfg(all(major_ge_3, minor_ge_1))]
+    pub fn is_derived_geographic(&self) -> bool {
+        unsafe { gdal_sys::OSRIsDerivedGeographic(self.0) == 1 }
+    }
+
+    #[inline]
+    pub fn is_local(&self) -> bool {
+        unsafe { gdal_sys::OSRIsLocal(self.0) == 1 }
+    }
+
+    #[inline]
+    pub fn is_projected(&self) -> bool {
+        unsafe { gdal_sys::OSRIsProjected(self.0) == 1 }
+    }
+
+    #[inline]
+    pub fn is_compound(&self) -> bool {
+        unsafe { gdal_sys::OSRIsCompound(self.0) == 1 }
+    }
+
+    #[inline]
+    pub fn is_geocentric(&self) -> bool {
+        unsafe { gdal_sys::OSRIsGeocentric(self.0) == 1 }
+    }
+
+    #[inline]
+    pub fn is_vertical(&self) -> bool {
+        unsafe { gdal_sys::OSRIsVertical(self.0) == 1 }
+    }
+
+    pub fn get_axis_orientation(&self, target_key: &str, axis: i32) -> AxisOrientationType {
+        // We can almost safely assume that if we fail to build a CString then the input
+        // is not a valide key.
```

Comment:
Should this be an Option/Result type? Or is it reasonable for users to infer that an error occurred when they get `gdal_sys::OGRAxisOrientation::OAO_Other`?

Reply:
you are right is should be a result type. The GDAL docs state that it will return "  the name of the axis or NULL on failure. "

Reply:
That was not obvious to me since  "No CPLError is issued on routine failures (such as not finding the AXIS)".

The error would then be a 'AxisNotFound' error value ?
In this case `get_axis_name` should also return a Result too not an `Option<String>` imho,

Reply:
Changed to Result return type for both `get_axis_orientation` and `get_axis_name`

Reply:
great thanks :+1: